### PR TITLE
add AlertBox2 iconOverride and onClose callback

### PR DESCRIPTION
**Pull request by jonjonray**
# Jira: [M5G-636](https://clever.atlassian.net/browse/M5G-636)

# Overview:
Adding two new props to AlertBox2 (which is in beta):
- onClose: which fires when the alert box is closed. We currently only manage that via local state and have no access to it from the consumer level
- iconOverride: design has a desire to customize the look of the icons on this component that don't always correspond to the defaults for success/critical/etc

Both props are optional, so shouldn't be a breaking change for existing consumers.

# Screenshots/GIFs:

# Testing:

- [ ] Unit tests
- Manual tests:
  - [x] Chrome
  - [ ] Safari
  - [ ] IE11

# Roll Out:
100%
- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT URL LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: docs/bundle.js.LICENSE.txt [hunk 1]
```diff
@@ -129,3 +138,7 @@ object-assign
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
+
+//! moment.js
+
+//! moment.js locale configuration
```
Comment (TheRealBenjiLi) [inline docs/bundle.js.LICENSE.txt:144]: What's with all this license updates? 😮

Comment (jonjonray) [reply docs/bundle.js.LICENSE.txt:144]: I feel like I usually get this whenever I run `make docs`

**Review by TheRealBenjiLi — APPROVED** [review]
Code itself looks good! Just had a question around the license text changes. If that is intentional, go ahead and merge 🌮

**Comment by jonjonray** [comment]
Not intentional, I can remove it and leave the rest!

**Pull request merged**